haystack/preview/dataclasses/answer.py

```diff
@@ -0,0 +1,23 @@
+from typing import List, Optional, Union
+from dataclasses import dataclass
+from haystack.preview.dataclasses.document import Document
+
+
+@dataclass
+class ExtractiveAnswer:
+    answer: str
+    start: int
+    end: int
+
+
+@dataclass
+class GenerativeAnswer:
+    answer: str
+
+
+@dataclass
+class Answer:
+    answer: Optional[Union[ExtractiveAnswer, GenerativeAnswer]]
```

Comment:
Wouldn't it better to have a generic `Answer` and have the others inherit from that?

Also let's avoid the repetition `Answer.answer`, it's really ugly and confusing.

Reply:
I thought the "no-answer" possibility was more explicit this way.
I originally also thought there were some issues with inheriting from dataclasses, but those only seem to be about default values.

Reply:
I also added some other reasoning to Sara's comment so it might make sense to continue the discussion there because you are both proposing a similar change.